src/structures/GuildScheduledEvent.js

```diff
@@ -0,0 +1,300 @@
+
+/**
+ * Represents a scheduled event in a {@link Guild}.
+ * @extends {Base}
+ */
+class GuildScheduledEvent extends Base {
+  /**
+   * @param {Client} client The instantiating client
+   * @param {APIGuildScheduledEvent} data The data for the guild scheduled event
+   */
+  constructor(client, data) {
+    super(client);
+
+    /**
+     * The id of the guild scheduled event
+     * @type {Snowflake}
+     */
+    this.id = data.id;
+
+    /**
+     * The id of the guild this guild scheduled event belongs to
+     * @type {Snowflake}
+     */
+    this.guildId = data.guild_id;
+
+    /**
+     * Whether the guild scheduled event has been deleted
+     * @type {boolean}
+     */
+    this.deleted = false;
+
+    this._patch(data);
+  }
+
+  _patch(data) {
+    /**
+     * The channel id in which the scheduled event will be hosted, or `null` if entity type is `EXTERNAL`
+     * @type {?Snowflake}
+     */
+    this.channelId = data.channel_id;
+
+    if ('creator_id' in data) {
+      /**
+       * The id of the user that created this guild scheduled event
+       * @type {Snowflake}
+       */
+      this.creatorId = data.creator_id;
+    }
+
+    /**
+     * The name of the guild scheduled event
+     * @type {string}
+     */
+    this.name = data.name;
+
+    if ('description' in data) {
+      /**
+       * The description of the guild scheduled event
+       * @type {string}
+       */
+      this.description = data.description;
+    }
+
+    /**
+     * The timestamp the guild scheduled event will start at
+     * @type {number}
+     */
+    this.scheduledStartTime = data.scheduled_start_time;
+
+    /**
+     * The timestamp the guild scheduled event will end at,
+     * or `null` if the event does not have a scheduled time to end
+     * @type {?number}
+     */
+    this.scheduledEndTime = data.scheduled_end_time;
+
+    /**
+     * The privacy level of the guild scheduled event
+     * @type {PrivacyLevel}
+     */
+    this.privacyLevel = GuildScheduledEventPrivacyLevels[data.privacy_level];
+
+    /**
+     * The status of the guild scheduled event
+     * @type {GuildScheduledEventStatus}
+     */
+    this.status = GuildScheduledEventStatuses[data.status];
+
+    /**
+     * The type of hosting entity associated with the scheduled event
+     * @type {GuildScheduledEventEntityType}
+     */
+    this.entityType = GuildScheduledEventEntityTypes[data.entity_type];
+
+    /**
+     * The id of the hosting entity associated with the scheduled event
+     * @type {?Snowflake}
+     */
+    this.entityId = data.entity_id;
+
+    if ('user_count' in data) {
+      /**
+       * The number of users who are subscribed to this guild scheduled event
+       * @type {number}
+       */
+      this.userCount = data.user_count;
+    }
+
+    if ('creator' in data) {
+      /**
+       * The user that created this guild scheduled event
+       * @type {User}
+       */
+      this.creator = this.client.users._add(data.creator);
+    }
```

Comment:
This is documented & typed as always available but is not sent when fetching an invite with an event id rendering this property `undefined`. It's also [documented at Discord](https://discord.com/developers/docs/resources/guild-scheduled-event#guild-scheduled-event-object-guild-scheduled-event-structure) as such. This should be nullable.

Reply:
I believe abaa35ad3af6a4e2ce1c2e95fa8765783b89930f fixed this and all the others like it